dpdispatcher/utils/dpcloudserver/client.py

```
@@ -274,10 +278,12 @@ def get_log(self, job_id):
         url, size = self._get_job_log(job_id)
         if not url:
             return ""
-        if self.last_log_offset >= size:
+        job_key = str(job_id)
+        offset = self.last_log_offsets.get(job_key, 0)
+        if offset >= size:
             return ""
-        resp = requests.get(url, headers={"Range": f"bytes={self.last_log_offset}-"})
-        self.last_log_offset += len(resp.content)
+        resp = requests.get(url, headers={"Range": f"bytes={offset}-"})
```

Ruff (0.15.21)

[error] Probable use of `requests` call without timeout

(S113)